### ISSUE TITLE
Crash Instead of showing a Missing Mods Screen

### ISSUE THREAD

**Issue by WenXin20** [delvr/Streams#27]
Streams v0.3
Farseek v2.0
Forge 2151
It crashes on game launch.
[Crash Log](http://pastebin.com/YLD4U9nY)

**[Edit]** 12/15/16
It happened when I deleted Forestry but forgot to delete the More Bees Mod.
So it seems like Streams is causing this crash instead of going to the Missing Mods warning page with the dirt background. After removing Streams and Farseek, instead of crashing it shows the Missing Mods warning.

**Comment by WesCook** [comment]
I'm experiencing the same crash with the new Streams/Farseek.  Not much to add other than it's not an isolated incident.

**Comment by ghost** [comment]
I had the same crash and found out it was caused by BiomePaintTools.

**Comment by WesCook** [comment]
I do not have BiomePaintTools installed, myself.

**Comment by ghost** [comment]
Ok, but WenXin20 does and I suggest to look there first. I can confirm the crash was gone after removing BPT from my modpack which has 71 other mods in it. Without BPT, it runs just fine. Maybe it's caused by several mods?

**Comment by WenXin20** [comment]
I removed Biome Paint Tools Mod and now it works for me too.

**Comment by WesCook** [comment]
Interesting, I may be the exception then.  I'll need to do some further testing to see if another mod is causing an issue here.

---

edit: I placed it back in my modpack environment and things seem to be working smoothly now...  I'm not sure what's changed.  Perhaps one of the mods I've discarded in the mean time.  Sorry for the confusion!

**Comment by ghost** [comment]
I figured out I forgot the Hammer Core mod. It's listed as a dependency but I missed it.
I don't know why Streams throws this error - a had the same as WenXin20: "java.lang.NoClassDefFoundError: streams/client/package" - but after putting Hammer Core in the mods folder, the crash isn't happening anymore.

**Comment by WenXin20** [comment]
That's odd, I already have the hammer core mod.

**Comment by canitzp** [comment]
This isn't a bug with any mod, it is a forge bug. This heppens if you have duplicated mods or missing mods. If you can show me your log file of the start, I've be able to tell you what does break it

**Comment by canitzp** [comment]
MinecraftForge/MinecraftForge#3501

**Comment by WenXin20** [comment]
I don't think I have any duplitcated or missing mods, but here is my log file:
[log](https://gist.github.com/WenXin20/35c17c368cc9f196adba808b043b7b75)

**Comment by canitzp** [comment]
this isn't the complete log file

**Comment by WenXin20** [comment]
Is the fml-client-latest log the one you wanted? Because that is what the log is, the game just named loaded past the construction stage before it crashed

**Comment by canitzp** [comment]
the log you posted is cut off, for me the last 1000 lines are important, they are enough, cause some uploader haven't infinite words/lines

**Comment by WenXin20** [comment]
Are you sure it's cut off, for me it's not, but at the top of the Web page it says that the file has been truncated, but that link leads to the full file: https://gist.githubusercontent.com/WenXin20/35c17c368cc9f196adba808b043b7b75/raw/995b291f222e0fbbf83776947d83b9ac0cea976f/Streams%2520Mod%2520Crash%2520w.%2520Biome%2520Paint%2520Tools%2520Latest%2520Log%2520

**Comment by canitzp** [comment]
here is your mistake. :03:56] [Client thread/ERROR] [FML/]: net.minecraftforge.fml.common.MissingModsException: Mod chickens (Chickens) requires [Forge@[12.18.3.2185,)]

**Comment by WenXin20** [comment]
Ok, thanks, I'll try updating forge to see if that fixes it tonight

**Comment by WenXin20** [comment]
Ok, that fixed it. Funny thing was that I had already updated forge, but I had  disabled the streams mod before I had updated forge. :D

**Comment by WenXin20** [comment]
Turns out that wasn't a real fix. I got the crash again, but this time it happened when I deleted Forestry but forgot to delete the More Bees Mod. 
So it seems like Streams is causing this crash instead of going to the Missing Mods warning page with the dirt background. After removing Streams and Farseek, instead of crashing it shows the Missing Mods warning.

**Comment by canitzp** [comment]
there is maybe a wrong class transforming that causes forge to load a non existing class, cause streams isn't loaded at this time, but it is scala code, so I have my problems to read it.

**Comment by delvr** [comment]
Thanks WenXin20, WesCook and canitzp for your help in investigating this so far. Sorry for the delay on my end, I'll look deeper into this issue and get back to you.

**Comment by canitzp** [comment]
No problem, I like to investigate such strange things

**Comment by WenXin20** [comment]
Cool, thanks for looking into it.

**Comment by WesCook** [comment]
What I've found of this error is that it shows up whenever there's a problem with loading Forge mods.  For instance, I accidentally had two versions of a mod installed.  Instead of throwing the usual Forge "duplicate mod ID" screen, it showed the "NoClassDefFoundError: streams/client/package" error.  Once I removed Streams/Farseek, I got the standard Forge error.

So I think in many cases Streams is being blamed where other mods are at fault; but at the same time, Streams is doing something to break the Forge error reporting.
